state_machine/drone.py

```diff
@@ -122,24 +122,23 @@ async def arm_and_takeoff(self, takeoff_alt: float) -> None:
         takeoff_alt: float
             Altitude to reach in meters
         """
+        logging.info("Waiting for vehicle to intialize...")
         while not self.vehicle.is_armable:
             # Vehicle is not ready to accept code
-            logging.info("Waiting for vehicle to initialize...")
+            await asyncio.sleep(0.5)
 
         self.vehicle.mode = dronekit.VehicleMode("GUIDED")
         self.vehicle.armed = True
 
         # Confirm vehicle is properly armed
-        while not self.vehicle.armed:
-            logging.info("Waiting for arming...")
-            await asyncio.sleep(1)
+        logging.info("Waiting for arming...")
+        while not self.vehicle.armed or self.vehicle.mode.name != "GUIDED":
+            await asyncio.sleep(0.5)
 
         self.vehicle.simple_takeoff(takeoff_alt)
```

Comment:
We might want to use a takeoff altitude slightly higher than the requested takeoff altitude to allow for some margin of error.